### DISCUSSION TITLE
findByCursor does not compute orderBy and populateOrderBy options correctly

### DISCUSSION THREAD

**Discussion by arthfael-breanainn** [mikro-orm/mikro-orm#6174] [General]
### Describe the bug

When I pass the orderBy parameter on my primaryKey by asking for a sort 'ASC', findByCursor returns me the expected result. As soon as I ask for a sort 'DESC' on this same primaryKey then findByCursor returns me an empty item array. Same if I want to sort my items by something other than the primaryKey.

In addition, the option populateOrderBy does not seem to be taken into account because the query generated by findByCursor never shows the sorting options

### Reproduction

this code :
```javascript
const response = await this.sampleRepository.findByCursor(
  { name: { $exists: true } },
      {
          populate: ['users', 'users.addresses'],
          filters,
          after: endCursor || { id: lastSeenId },
          orderBy: { name: QueryOrder.ASC },
          populateOrderBy: {
            users: { name: QueryOrder.ASC, addresses: { street: QueryOrder.ASC } },
          },
          first: limit,
      },
  );
```

produce this output :
```SQL
select 
  "s0".*, 
  "u1"."id" as "u1__id", 
  "u1"."name" as "u1__name", 
  "u1"."sample_id" as "u1__sample_id", 
  "a2"."id" as "a2__id", 
  "a2"."street" as "a2__street", 
  "a2"."user_id" as "a2__user_id" 
from "sample" as "s0" 
left join "user" as "u1" on "s0"."id" = "u1"."sample_id" 
left join "address" as "a2" on "u1"."id" = "a2"."user_id" 
where "s0"."id" in (
  select 
    "s0"."id" 
  from (
    select 
      "s0"."id" 
    from "sample" as "s0" 
    left join "user" as "u1" on "s0"."id" = "u1"."sample_id" 
    left join "address" as "a2" on "u1"."id" = "a2"."user_id" 
    where "s0"."name" is not null and "s0"."name" > NULL 
    group by "s0"."id" 
    order by min("s0"."name") asc 
    limit 11
  ) as "s0"
) 
order by "s0"."name" asc

select count(*) as "count" from "sample" as "s0" where "s0"."name" is not null
```

I received this :
```JSON
{
  "data": {
    "samples": {
      "items": [],
      "startCursor": null,
      "endCursor": null,
      "hasPrevPage": true,
      "hasNextPage": false,
      "length": 0,
      "totalCount": 21
    }
  }
}
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

^6.3.13

### Node.js version

v20.17.0

### Operating system

alpine

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Please provide a [complete reproduction](https://github.com/mikro-orm/reproduction), something I can run myself to see the problem.

**Reply by arthfael-breanainn** [reply]
Sorry for the inconvenience, I tried again today and finally everything works. It’s a mistake on my part, I had not read the doc well!
I feel really stupid. Again sorry for the inconvenience!

**Reply by B4nan** [reply]
No worries, bugs happen, but they are often about some tiny details, so having a complete repro is important.
